src/Mvc/Mvc.Core/src/Formatters/TextOutputFormatter.cs

```diff
@@ -132,7 +132,7 @@ public override Task WriteAsync(OutputFormatterWriteContext context)
 
             if (context.HttpContext.RequestServices.GetService<IProblemDetailsService>() is { } problemDetailsService)
             {
-                return problemDetailsService.WriteAsync(new ()
+                return problemDetailsService.TryWriteAsync(new ()
```

Comment:
What happens if it can't?

Reply:
It will do what it was doing before, only setting the status code.